src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -1,50 +1,59 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.model.Model;
+import seedu.address.model.person.ModuleRoleContainsKeywordsPredicate;
 import seedu.address.model.person.NameContainsKeywordsPredicate;
+import seedu.address.model.person.Person;
 
 
 /**
- * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Finds and lists all persons in address book whose name or module contains any of the argument keywords.
  * Keyword matching is case insensitive.
  */
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names or modules contain "
+            + "any of the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: "
             + PREFIX_NAME + "KEYWORD ["
-            + PREFIX_NAME + "MORE_KEYWORDS]...\n"
+            + PREFIX_NAME + "MORE_KEYWORDS]..."
+            + PREFIX_MODULE + "KEYWORD ["
+            + PREFIX_MODULE + "MORE_KEYWORDS]...\n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "alice "
             + PREFIX_NAME + "bob "
-            + PREFIX_NAME + "charlie";
+            + PREFIX_MODULE + "CS1101S"
+            + PREFIX_MODULE + "CS2030S";
 
-    private final NameContainsKeywordsPredicate predicate;
+    private final List<Predicate<Person>> predicates;
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    public FindCommand(List<Predicate<Person>> predicates) {
+        this.predicates = predicates;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+        predicates.forEach(model::updateFilteredPersonList);
+        String formattedKeywords = getAllKeywordsFromPredicates(predicates);
+
```

Comment:
I think the reason why the find command only works for 1 predicate (although I'm not sure why it only works for module but not name) is the way that predicates being passed into the ``model::updateFilteredPersonList``. Maybe you can try to combine the predicates before passing it into the method?

Reply:
![image](https://github.com/user-attachments/assets/9d217a63-3345-4ec3-8e30-5d6871df89e2)
Maybe Predicate::or could be useful here

Reply:
Thank you for pointing out! I think this is the correct way of doing it!